### DISCUSSION TITLE
Any plans to update the UI?

### DISCUSSION THREAD

**Discussion by tomhickerson** [reliatec-gmbh/LibreClinica#220] [Q&A] [answered]
I was wondering if Reliatec had anything on the roadmap to update/upgrade the UI?  There are some older libraries running the jQuery/JMesa UI which could use an upgrade...

**Comment by toskrip** [comment, marked as answer]
Hi, just want to clarify that LibreClinica is community project and Reliatec is a valuable member of the community (providing lot of organisatory and technical background). In no way I am speaking for Reliatec, however change as big as the update of frontend for LibreClinica is not made by one member alone and there has to be an agreement across main collaborators. We have a regular telco meetings where current developments and further directions are discussed (if you are interested, it should not be a problem to invite you). The topic of UI was present and although we agreed it is important, it is also huge (maybe jquery update is possible but jmesa is quite a beast). Recent affords go more towards backend cleanup. So far no collaboration party came with a clear idea about how to proceed with UI updates so there is no concrete plan approved yet (not even proposed). The rule of thumb is that if somebody steps in with an idea, we would like that party to be able to substantially contribute (developer speaking) in making that idea reality. Ideally the development of LibreClinica should be more de-centralised, but still keeping the focus testing the contributions that goes to releases to not break any fundamental system features.

best

Tomas

**Reply by tomhickerson** [reply]
Hi Tomas,

Thanks very much for letting me know about this.  I'd definitely like to be included in future telcos, when possible.

I agree that jQuery would be the biggest part of the UI upgrade.  There are smaller pieces that can be divided out in parallel with that, including removal/replacement of older or defunct libraries (JDom and Sitemesh come to mind), and then removal/replacement parts of the UI which are now extraneous (removal of the footer, changes to the CSS, removal of older parts of HTML). 

I agree that JMesa is 'embedded' enough to be not a small task to change, so that may have to stay in the code base for a while yet.

Thanks again, Tom

**Reply by toskrip** [reply]
sent you email with telco details.
